app/controllers/container_node_controller.rb

```diff
@@ -6,4 +6,19 @@ class ContainerNodeController < ApplicationController
   after_action :cleanup_action
   after_action :set_session_data
 
+  def launch_cockpit
+    node = identify_record(params[:id], ContainerNode)
+
+    if node.ipaddress
+      url = node.cockpit_url
```

Comment:
I'd say that lines 13 to  23 can be extracted to a method that would be shared with https://github.com/ManageIQ/manageiq/pull/10083/files#diff-5910cce3bec4cd624ba288de86215b4cR93